executable/executable.go

```diff
@@ -213,6 +221,7 @@ func (e *Executable) RunWithStdin(stdin []byte, args ...string) (ExecutableResul
 // Wait waits for the program to finish and results the result
 func (e *Executable) Wait() (ExecutableResult, error) {
 	defer func() {
+		e.cancelFunc()
```

Comment:
We don't need this - confirm

Reply:
Let's reset `ctxWithTimeout` to `nil`, just like all other props

Reply:
@rohitpaulk  When trying to get rid of `cancelFunc`, LSP gave me a warning. Should I ignore it?

`ctx, _ := context.WithTimeout(...)`

the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak

> lostcancel: check cancel func returned by context.WithCancel is called

> The cancellation function returned by context.WithCancel, WithTimeout, and WithDeadline must be called or the new context will remain live until its parent context is cancelled. (The background context is never cancelled.)

https://pkg.go.dev/golang.org/x/tools/go/analysis/passes/lostcancel